File: app/src/main/java/knf/kuma/directory/DirPagerAdapter.java
package knf.kuma.directory;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

public class DirPagerAdapter extends FragmentPagerAdapter {

    private DirectoryPageFragment animes=DirectoryPageFragment.get(DirectoryPageFragment.DirType.ANIMES);
    private DirectoryPageFragment ovas=DirectoryPageFragment.get(DirectoryPageFragment.DirType.OVAS);
    private DirectoryPageFragment movies=DirectoryPageFragment.get(DirectoryPageFragment.DirType.MOVIES);

    public DirPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        switch (position){
            default:
            case 0:
                return "ANIME";
            case 1:
                return "OVA";
            case 2:
                return "PELICULA";
        }
    }

    public void onChangeOrder(){
        animes.onChangeOrder();
        ovas.onChangeOrder();
        movies.onChangeOrder();
    }

    @Override
    public Fragment getItem(int position) {
        switch (position){
            default:
            case 0:
                return animes;
            case 1:
                return ovas;
            case 2:
                return movies;
        }
    }

    @Override
    public int getCount() {
        return 3;
    }
}

File: app/src/main/java/knf/kuma/directory/DirPagerAdapter.kt
package knf.kuma.directory

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

class DirPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    private val animes = DirectoryPageFragment[DirectoryPageFragment.DirType.ANIMES]
    private val ovas = DirectoryPageFragment[DirectoryPageFragment.DirType.OVAS]
    private val movies = DirectoryPageFragment[DirectoryPageFragment.DirType.MOVIES]

    override fun getPageTitle(position: Int): CharSequence? {
        return when (position) {
            1 -> "OVA"
            2 -> "PELICULA"
            else -> "ANIME"
        }
    }

    fun onChangeOrder() {
        animes.onChangeOrder()
        ovas.onChangeOrder()
        movies.onChangeOrder()
    }

    override fun getItem(position: Int): Fragment {
        return when (position) {
            1 -> ovas
            2 -> movies
            else -> animes
        }
    }

    override fun getCount(): Int {
        return 3
    }
}
